src/safeposix/syscalls/fs_calls.rs

```diff
@@ -1726,4 +1726,22 @@ impl Cage {
             syscall_error(Errno::EBADF, "getdents", "Invalid file descriptor")
         }
     }
+
+    //------------------------------------GETCWD SYSCALL------------------------------------
+    
+    pub fn getcwd_syscall(&self, buf: *mut u8, bufsize: u32) -> i32 {
+        let cwd: String = format!("{}{}", self.cwd.read().unwrap().to_string_lossy(), "\0");
+        let bytes: Vec<u8> = cwd.as_bytes().to_vec();
+        let length = bytes.len();
+
+        use std::convert::TryInto;
```

Comment:
We try very hard to import as little as possible even from the rust standard library and when we do we make sure to do it through the interface.

Reply:
Fixed with the last update.